app/controllers/course/survey/responses_controller.rb

```diff
@@ -27,7 +28,11 @@ def show
     render 'course/survey/surveys/index'
   end
 
+  # This action returns data that will be displayed the show page. Since the updates made
+  # are not specified by the user, we only ensure that user has permission to
+  # :read the response.
   def edit
+    authorize!(:read, @response)
```

Comment:
@allenwq suggested that for react pages, `show`, `index`, `create` and `destroy` should only contain 'typical' behaviour. Other actions including `edit` can be used to handle special behaviour. In this case, we don't want a request to `show` to be adding rows to the DB.

What do you think if I moved this to another action, e.g. `show_latest`?